### PR TITLE
Feature/sectioned form

### PR DESCRIPTION
**Pull request by emclaug2**
#### Changes proposed in this Pull Request:
- Add sectioned form example

### CODE REVIEW

**Comment by huayunh** [comment]
image

- [ ] Can you make this dropdown text field pre-selected
- [ ] Can you make this dropdown text field show its label
- [ ] Is the font size in the tooltip pre-determined by the theme? If so, can you log an enhancement to make the font size 12dp?

**Comment by huayunh** [comment]
Update: strange... in CSB the list status is as described

**Comment by emclaug2** [comment]
> Update: strange... in [CSB](https://codesandbox.io/s/github/pxblue/react-design-patterns/tree/feature/sectioned-form) the list status is as described

This is how I see it locally.

**Comment by emclaug2** [comment]
> * [ ]  Is the font size in the tooltip pre-determined by the theme? If so, can you log an enhancement to make the font size 12dp?

Yep this is the default tooltip size, I'll log an issue for this.

**Comment by huayunh** [comment]
fresh cloned and worked. Weird.

**Comment by huayunh** [comment]
May want to submit an issue for this too:
image

To reproduce:
- Click in the text field
- without selecting anything, click outside the menu
- Observe

**Comment by huayunh** [comment]
Functionality-wise, everything looks good to me.

Visually, the icons next to the subsection headings should be gray500. And those dividers shall protrude out. 

image

**Comment by emclaug2** [comment]
I've updated the icon to be Gray[500] using `theme.palette.text.second` & have adjusted the dividers width so it protrudes out a bit.

@huayunh I cannot reproduce the error you're seeing though, here's what I see: 

WhereEr

File: src/pages/form-validation/sectioned-form/index.tsx [hunk 1]
```diff
@@ -0,0 +1,496 @@
...
+                            >
+                                <MenuItem value={'level I'}>Level I (Regional)</MenuItem>
+                                <MenuItem value={'level II'}>Level II (Regional)</MenuItem>
+                                <MenuItem value={'level III'}>Level III (Regional)</MenuItem>
+                            </Select>
+                        </FormControl>
+                        <Hidden xsDown>
+                            <FormControlLabel control={<Checkbox name="checkedC" />} label={pxbProtection} />
+                            <Tooltip title={pxbProtectionDescription} arrow placement={'top'}>
+                                <HelpOutline style={{ color: 'rgba(66, 78, 84, .3)' }} />
```
Comment (huayunh) [inline src/pages/form-validation/sectioned-form/index.tsx:309]: theme.palette.text.disabled

File: src/pages/form-validation/sectioned-form/index.tsx [hunk 2]
```diff
@@ -0,0 +1,496 @@
...
+                                }}
+                            >
+                                <FormControlLabel
+                                    control={<Checkbox name="checkedC" />}
+                                    label={''}
+                                    style={{ marginTop: -6 }}
+                                />
+                                <div>
+                                    <Typography variant={'body1'}>PXB Protection</Typography>
+                                    <Typography variant={'body2'} style={{ color: Colors.gray[500] }}>
```
Comment (huayunh) [inline src/pages/form-validation/sectioned-form/index.tsx:328]: theme.palette.text.secondary

File: src/pages/form-validation/sectioned-form/index.tsx [hunk 3]
```diff
@@ -0,0 +1,496 @@
...
+                setEmailError('');
+            }
+        },
+        [setEmail, emailError, validateEmail, setEmailError]
+    );
+
+    const characterLimitsHelperText = (
+        <>
+            <span>{!name && showRequiredError ? 'Required' : 'For example, Facility or Campus name'}</span>
+            <span style={{ float: 'right' }}>{`${name.length}/${MAX_CHARS_LIMIT}`}</span>
```
Comment (huayunh) [inline src/pages/form-validation/sectioned-form/index.tsx:190]: Can we make this stay theme.palette.text.secondary, even on error? Like this:

image

**Review by huayunh — APPROVED** [review]

**Comment by emclaug2** [comment]
Thanks for the approval on this PR @huayunh  I'll update the colors and then merge it in.

**Pull request merged**